igvc_rviz_plugins/src/WaypointPanel.cpp

```diff
@@ -0,0 +1,56 @@
+#define _USE_MATH_DEFINES
+
+#include <igvc_rviz_plugins/WaypointPanel.h>
+#include <QVBoxLayout>
+#include <pluginlib/class_list_macros.h>
+
+namespace igvc_rviz_plugins {
+
+WaypointPanel::WaypointPanel(QWidget *parent)
+  : rviz::Panel(parent)
+{
+    ros::NodeHandle handle;
+
+    QLabel *label = new QLabel("0");
+
+    waypoint_subscriber = handle.subscribe<geometry_msgs::PointStamped>("/waypoint", 1, boost::bind(&WaypointPanel::waypoint_callback, this, _1, label));
+    robot_position_subscriber = handle.subscribe<nav_msgs::Odometry>("/odometry/filtered", 1, boost::bind(&WaypointPanel::robot_position_callback, this, _1, label));
+
+    QVBoxLayout *layout = new QVBoxLayout;
+    layout->addWidget(label);
+    setLayout(layout);
+}
+
+void WaypointPanel::waypoint_callback(const geometry_msgs::PointStampedConstPtr &msg, QLabel *label) {
+    way_x = msg->point.x - robot_x;
+    way_y = msg->point.y - robot_y;
+}
+
+void WaypointPanel::robot_position_callback(const nav_msgs::OdometryConstPtr &msg, QLabel *label) {
+    double distance = sqrt(pow(way_x, 2) + pow(way_y, 2));
```

Comment:
Consider replacing with a call to [std::hypot](http://en.cppreference.com/w/cpp/numeric/math/hypot)